lib/date_time/date_time.c

```diff
@@ -165,9 +165,6 @@ static int date_time_init(void)
 	return 0;
 }
 
-/* Initialization should be before lte_lc (uses CONFIG_APPLICATION_INIT_PRIORITY)
- * so that we can subscribe to xtime notification if CONFIG_LTE_AUTO_INIT_AND_CONNECT is enabled
- */
 #define DATE_TIME_INIT_PRIO 80
 
 SYS_INIT(date_time_init, APPLICATION, DATE_TIME_INIT_PRIO);
```

Comment:
Well, I removed that because connecting to the LTE network can't happen during `SYS_INIT` anymore, so if this code runs at `SYS_INIT` it will always run before the application has connected to the network. Plus registering `lte_lc` event handlers can happen at any time (no initialization requirements anymore), so this we can successfully register them here and be ready to receive the first event.

Reply:
Ok, in that case I think we should set some standard priority for date_time library. E.g., CONFIG_APPLICATION_INIT_PRIORITY.

Reply:
OK, I can update that.

Reply:
I have created a new commit for this change